File: Server/io/Message.cs
namespace Gopet.IO
{
    public class Message
    {

        public sbyte id;

        private DataOutputStream dos;
        private DataInputStream dis;
        public bool isEncrypted;
        public static bool isiWin = false;

        public Message(int command) : this(command, false)
        {

        }

        public Message(int command, bool isEncrypted)
        {
            this.isEncrypted = false;
            id = (sbyte)(command & 0xFF);
            this.isEncrypted = isEncrypted;
        }

        public Message(sbyte[] data)
        {
            isEncrypted = false;
            sbyte[] msgData = new sbyte[data.Length - 1];
            Buffer.BlockCopy(data, 1, msgData, 0, msgData.Length);
            id = data[0];
            dis = new DataInputStream(msgData);
        }

        public sbyte[] getBuffer()
        {
            if (dos == null)
            {
                return new sbyte[] { id };
            }
            else
            {
                sbyte[] data = dos.getData();
                sbyte[] buffer;
                if (isiWin)
                {
                    buffer = new sbyte[data.Length + 3];
                    buffer[0] = 40;
                    buffer[1] = (sbyte)(id >>> 8 & 255);
                    buffer[2] = (sbyte)(id >>> 0 & 255);
                    Buffer.BlockCopy(data, 0, buffer, 3, data.Length);
                    return buffer;
                }
                else
                {
                    buffer = new sbyte[data.Length + 1];
                    buffer[0] = id;
                    Buffer.BlockCopy(data, 0, buffer, 1, data.Length);
                    return buffer;
                }
            }
        }

        public DataInputStream reader()
        {
            return dis;
        }

        public DataOutputStream writer()
        {
            if (dos == null)
            {
                dos = new DataOutputStream();
            }

            return dos;
        }

        public void putsbyte(int value)
        {

            writer().writeByte(value);

        }

        public void putString(string text)
        {

            writer().writeUTF(text);

        }

        public void putUTF(string text)
        {

            writer().writeUTF(text);

        }

        public void putInt(int value)
        {

            writer().writeInt(value);

        }

        public void putbool(bool value)
        {

            writer().writeBool(value);

        }

        public void putShort(int value)
        {

            writer().writeShort(value);

        }

        public void putlong(long value)
        {

            writer().writeLong(value);

        }

        public void cleanup()
        {
            try
            {
                if (dis != null)
                {
                    dis.Close();
                }

                if (dos != null)
                {
                    dos.Close();
                }
            }
            catch (IOException var2)
            {
            }
        }

        public sbyte readsbyte()
        {
            return reader().readsbyte();
        }

        public short readShort()
        {
            return reader().readShort();
        }

        public int readUnsignedsbyte()
        {
            return reader().readUnsignedByte();
        }

        public int readUnsignedShort()
        {
            return reader().readUnsignedShort();
        }

        public string readUTF()
        {
            return reader().readUTF();
        }

        public long readlong()
        {
            return reader().readlong();
        }





        public int readInt()
        {
            return reader().readInt();
        }

        public void Close()
        {
            cleanup();
        }
    }
}

File: goPetJ2ME/src/vn/me/network/Message.java
package vn.me.network;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

/* renamed from: Class159  reason: default package */
 /* loaded from: gopet_repackage.jar:message.class */
public final class Message {

    public int id;
    private ByteArrayOutputStream baos;
    private DataOutputStream dos;
    private DataInputStream dis;
    public boolean isEncrypted;
    public static boolean isiWin = false;

    public Message(int i) {
        this(i, true);
    }

    public Message(int i, boolean z) {
        this.isEncrypted = true;
        this.id = i;
        this.isEncrypted = true;
    }

    public Message(byte[] bArr) {
        this.isEncrypted = true;
        byte[] bArr2 = new byte[bArr.length - 1];
        System.arraycopy(bArr, 1, bArr2, 0, bArr2.length);
        this.id = bArr[0];
        this.dis = new DataInputStream(new ByteArrayInputStream(bArr2));
    }

    public final byte[] getBuffer() {
        if (this.baos == null) {
            return new byte[]{(byte) this.id};
        }
        byte[] byteArray = this.baos.toByteArray();
        byte[] bArr = new byte[byteArray.length + 1];
        bArr[0] = (byte) this.id;
        System.arraycopy(byteArray, 0, bArr, 1, byteArray.length);
        return bArr;
    }

    public final DataInputStream reader() {
        return this.dis;
    }

    private DataOutputStream writer() {
        if (this.baos == null) {
            this.baos = new ByteArrayOutputStream();
            this.dos = new DataOutputStream(this.baos);
        }
        return this.dos;
    }

    public final void putByte(int i) {
        try {
            writer().writeByte(i);
        } catch (IOException unused) {
        }
    }

    public final void putString(String str) {
        try {
            writer().writeUTF(str);
        } catch (IOException unused) {
        }
    }

    public final void putInt(int i) {
        try {
            writer().writeInt(i);
        } catch (IOException unused) {
        }
    }

    public final void putBoolean(boolean z) {
        try {
            writer().writeBoolean(z);
        } catch (IOException unused) {
        }
    }

    public void putShort(int value) {
        try {
            this.writer().writeShort(value);
        } catch (IOException var3) {
        }

    }

    public final void cleanup() {
        try {
            if (this.dis != null) {
                this.dis.close();
            }
            if (this.dos != null) {
                this.dos.close();
            }
        } catch (IOException unused) {
        }
    }
}
